src/cluster/placement/algo/sharded_test.go

```diff
@@ -1306,6 +1306,42 @@ func TestBalanceShardsForShardedWhenImbalanced(t *testing.T) {
 	assert.Equal(t, expectedInstances, balancedPlacement.Instances())
 }
 
+func TestInitialPlacementIsBalanced(t *testing.T) {
+	instances := make([]placement.Instance, 0)
+
+	for i := 0; i < 10; i++ {
+		instances = append(instances,
+			placement.NewEmptyInstance(fmt.Sprintf("instance-0-%03d", i), "iso0", "zone", "endpoint", 1),
+			placement.NewEmptyInstance(fmt.Sprintf("instance-1-%03d", i), "iso1", "zone", "endpoint", 1),
+		)
+	}
+
+	shardIDs := make([]uint32, 1024)
+	for i := range shardIDs {
+		shardIDs[i] = uint32(i)
+	}
+
+	algo := newShardedAlgorithm(placement.NewOptions())
+	p, err := algo.InitialPlacement(instances, shardIDs, 2)
+	require.NoError(t, err)
+
+	var (
+		min = math.MaxInt32
+		max = math.MinInt32
+	)
+	for _, instance := range p.Instances() {
+		n := instance.Shards().NumShards()
+		if n < min {
+			min = n
+		}
+		if n > max {
+			max = n
+		}
+	}
+	require.Equal(t, 102, min)
+	require.Equal(t, 103, max)
```

Comment:
Nit: just do the range assertions in the loop and drop 10 LoC.

Reply:
Migrated to property test instead: https://github.com/m3db/m3/pull/4020/commits/ebc054bfa2f292d5b10e0115fd035e65f0fda888